### PR TITLE
Factor out all the little thread pools.

### PR DESCRIPTION
**Pull request by hgschmie**
Factor thread pool management and control into guice injected, managed pools.

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java [hunk 1]
```diff
@@ -135,6 +135,10 @@
 
   private long zookeeperAsyncTimeout = 5000;
 
+  private int coreThreadpoolSize = 8;
```
Comment (hgschmie) [inline SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java]: Sharing them allows to factor the thread pool management out of a lot of components which in turn no longer need to be managed (only managed thing was the threadpool shutdown). The "8" is a magic number based on the number of thread pools in existence. We have a metric which tracks the number of threads actually used and I may adjust later.

Comment (wsorenson) [inline SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java]: Could we just have some sort of thread pool provider that is managed?

Comment (hgschmie) [inline SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java]: You mean like a SingularityManagedScheduledExecutorServiceProvider? :-)

Comment (wsorenson) [inline SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java]: I was thinking we should have it be a Provider like a Factory that creates a new Managed Executor Service. That way we don't have to hard code the # of threads, and it also guarantees that each user can not mistakenly over-use threads in the pool and starve another process.

**Comment by wsorenson** [comment]
Addressed in #417

**Pull request closed without merging**